Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java

```diff
@@ -1152,7 +1154,7 @@ void extrapolateBasicSliding(RangeVector result, int horizon, int blockSize, flo
             System.arraycopy(queryPoint, blockSize, queryPoint, 0, dimensions - blockSize);
 
             SampleSummary imputedSummary = getConditionalFieldSummary(queryPoint, blockSize, missingIndexes, 1, 0,
-                    false, false, centrality);
+                    false, false, centrality, 1);
```

Comment:
Lots of great discussion points. First, what is intended -- the shingleSize 1 means we will compute statistics for the entire vector and output that. ShingleSize 2 means that we would only compute the statistic of the second (last) half of the vector. Likewise 3, 4 ... etc. Keep in mind for forecasting the shingling size is large(ish, specially if the horizon is large) so doing this would save a lot of time (and indeed the current PR seems to save 60%+ time for the build for parkservices :)

Now for the ideas/point you raised -- we actually do some filtering based on the centrality parameter -- centrality = 1 will throw away the far points. Centrality = 0 will not perform any filtering. *Now* we could be using the using the distance (lets see harmonic mean/something) as weights -- that would be reasonable and better in some scenarios. But I have not managed to experiment much -- in the small examples I saw, I found that changing the weights was introducing variance/non-smoothness in the result. But I am sure it can be controlled. One other idea worth using is to use (inverse of) RCF scores themselves as weight. Same issue - the results are better in some cases, but then needs more controlling and experiments.